### PR TITLE
Fix: Timezone dependence during tests

### PR DESCRIPTION
**Pull request by bornova**
Hi @nicolashefti,
I tried to take a stab at this [pesky timezone issue](https://github.com/formulajs/formulajs/issues/85).  I came up with a fix that works but feels a bit hacky.

Dates in test cases are either `yyyy-mm-dd` or `mm/dd/yyyy` format.  Unfortunately, since `new Date()` creates a date object in user's local timezone, this creates issues.  For example, since I am near Los Angeles, the following dates give different results.
```javascript
new Date('2007-01-01')  // Sun Dec 31 2006 16:00:00 GMT-0800 (Pacific Standard Time)
```
```javascript
new Date('1/1/2007')  // Mon Jan 01 2007 00:00:00 GMT-0800 (Pacific Standard Time)
```

In order to address this, I decided to modify the date string if it matched the format` yyyy-mm-dd`, by adding `T00:00:00.000`:
https://github.com/formulajs/formulajs/blob/8835f778120ca6008ab056e8a104abab9e5e6ee6/src/utils/common.js#L232 because following is timezone independent:
```javasript
new Date('2007-01-01T00:00:00.000')  // Mon Jan 01 2007 00:00:00 GMT-0800 (Pacific Standard Time)
```
I also changed the date returned by `serialNumberToDate()` function to UTC time:
https://github.com/formulajs/formulajs/blob/8835f778120ca6008ab056e8a104abab9e5e6ee6/src/utils/common.js#L205-L206
https://github.com/formulajs/formulajs/blob/8835f778120ca6008ab056e8a104abab9e5e6ee6/src/utils/common.js#L213

I can't guarantee there won't be any edge cases, but I tested using several [timezones](https://en.wikipedia.org/wiki/List_of_tz_database_time_zones) and all unit-tests pass.  I know Formula.js is not a date parsing library and adding an external library such as Moment.js (or now Luxon) may not be practical, maybe this would be a good patch?  Let me know what you think.

### CODE REVIEW

**Comment by bornova** [comment]
Hi @nicolashefti,
I know you have #88 WIP as well but any thoughts on this?  Do you think it's a viable patch or should I close it?

**Comment by nicolashefti** [comment]
Hi @bornova. I agree, we should move on this topic. I'll check that this week.

**Comment by nicolashefti** [comment]
I think your approach is good, even if it is not bullet proof. On my PR #88 I chose to change the tests which failed depending on TZ. I think we can move on with your version.

**Review by nicolashefti — COMMENTED** [review]
One small detail, otherwise ready to go for me.

File: src/utils/common.js [hunk 1]
```diff
@@ -229,7 +229,7 @@ export function parseDate(date) {
   }
 
   if (typeof date === 'string') {
-    date = new Date(date)
+    date = date.match(/(\d{4})-(\d\d?)-(\d\d?)$/) ? new Date(date + 'T00:00:00.000') : new Date(date)
```
[the remark this answers is not in the document]
Comment (bornova) [reply src/utils/common.js:232]: Agreed.  I will make the change.

**Review by nicolashefti — APPROVED** [review]
